elbepack/initvmaction.py

```diff
@@ -234,15 +323,49 @@ class StopAction(InitVMAction):
     def __init__(self, node):
         InitVMAction.__init__(self, node)
 
-    def execute(self, _initvmdir, _opt, _args):
+    def _stop_qemu_vm(self, initvmdir):
+        """
+        Stop the QEMU initvm.
+
+        This method tries to stop the QEMU initvm by sending a poweroff event
+        using QEMU monitor.
+        """
+        initvmdir = check_initvm_dir(initvmdir)
+
+        socket_path = os.path.join(initvmdir, 'qemu-monitor-socket')
+
+        # Test if QEMU monitor unix-socket file exists, and error exit if not.
+        if not os.path.exists(socket_path):
+            print('No unix socket found for this vm!\nunable to shutdown this vm.')
+            sys.exit(212)
+
+        try:
+            with socket.socket(socket.AF_UNIX, socket.SOCK_STREAM) as client:
+                client.connect(socket_path)
+                client.sendall("system_powerdown\n".encode('utf-8'))
```

Comment:
`client.sendall(b"system_powerdown\n")`

(I realize that I asked for the `'utf-8'` before, but the binary literal is even better)